x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/AuthenticationServiceTests.java

```diff
@@ -340,8 +343,7 @@ public void testAuthenticateBothSupportSecondSucceeds() throws Exception {
         }, this::logAndFail));
         assertTrue(completed.get());
         verify(auditTrail).authenticationFailed(reqId, firstRealm.name(), token, "_action", transportRequest);
-        verify(realms).asList();
-        verifyNoMoreInteractions(realms);
+        verify(realms, atLeastOnce()).asList();
```

Comment:
Should it? Testing how many times a method is called tends to be an anti-pattern. Why do we can if it was called once or twice?

Reply:
The concern is different here because `realms.asList()` is called once in `init()` as:
```java
assertThat(this.realms.asList(), equalTo(realmList));
```
So either we don't test for this interaction at all. Or we should test for at least twice because the first time is done as part of test setup and there is no piont to test it.